src/main/java/com/alkemy/ong/controllers/OrganizationController.java

```diff
@@ -1,32 +1,52 @@
 package com.alkemy.ong.controllers;
 
 import com.alkemy.ong.dto.OrganizationDTO;
+import com.alkemy.ong.dto.ReducedOrganizationDTO;
 import com.alkemy.ong.services.OrganizationService;
+import com.alkemy.ong.utility.GlobalConstants;
+import javassist.NotFoundException;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
+import javax.validation.Valid;
+import java.io.IOException;
 import java.util.List;
 
+//@PreAuthorize("permitAll()") // Descomentar cuando se trabaje en el ticket 73
 @RestController
-@RequestMapping("/organization/public")
+@RequestMapping(GlobalConstants.Endpoints.ORGANIZATION_PUBLIC_INFO)
 public class OrganizationController {
 
     @Autowired
     private OrganizationService organizationService;
 
     @GetMapping
-    public ResponseEntity<List<OrganizationDTO>> getAll() {
-        List<OrganizationDTO> dtos = organizationService.getAll();
+    public ResponseEntity<List<ReducedOrganizationDTO>> getAll() {
+        List<ReducedOrganizationDTO> dtos = organizationService.getAll();
         return ResponseEntity.ok().body(dtos);
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<OrganizationDTO> getById(@PathVariable String id) {
-        OrganizationDTO dto = organizationService.getById(id);
+    public ResponseEntity<ReducedOrganizationDTO> getById(@PathVariable String id) {
+        ReducedOrganizationDTO dto = organizationService.getById(id);
         return ResponseEntity.ok().body(dto);
     }
+
+    //@PreAuthorize("hasRole('ADMIN')") // Descomentar cuando se trabaje en el ticket 73
+    @PostMapping
+    public ResponseEntity<OrganizationDTO> updateOrganization(@RequestParam(value = "file", required = false) MultipartFile image,
+                                                              @ModelAttribute OrganizationDTO organizationDTO){
+
+        try{
+            OrganizationDTO org = organizationService.updateOrganization(image, organizationDTO);
+
+            return ResponseEntity.ok().body(org);
+        } catch (IOException e){
```

Comment:
Si bien `amazonService.uploadFile()` dice en el prototipo q tira `IOException`, si te fijás en la doc de la implementación y en el ejemplo del controller, tira 2, se marca como q tira IOException sola porq las dos excepciones heredan de esa. Por otro lado ninguna de esas dos excepciones indica un not found, q es un error q un update debería controlar así q a lo mejor te convendría tira una exception q indique eso en la implementación de `updateOrganizacion()`.